### PR TITLE
CON-2661 [CHECKPOINT-RESTRUCTURE] Make `gcd` solution a function instead of a program

### PR DESCRIPTION
**Pull request by nprimo**
- update tests accordingly

### CODE REVIEW

**Review by MSilva95 — CHANGES_REQUESTED** [review]
The test is not responding when we give wrong solutions

File: tests/gcd_test/main.go [hunk 1]
```diff
@@ -1,30 +1,32 @@
 package main
```
[the remark this answers is not in the document]
Comment (nprimo) [reply tests/gcd_test/main.go:1]: I got a timeout when testing with this solution:

```console
timeout: Did you write an infinite loop? context deadline exceeded%
```

**Review by MSilva95 — APPROVED** [review]
LGTM!

**Review by HarryVasanth — APPROVED** [review]
> [!NOTE]
>  ✅ Production and pre-production prerequisites have been deployed.
> #### Clear to merge!

**Pull request merged**